components/AgentForm.tsx

```diff
@@ -193,6 +208,23 @@ export default function BaseForm(props: Props) {
     Prisma.PromiseReturnType<typeof getDatastores>
   >('/api/datastores', fetcher);
 
+
+  const handleSubmitRateLimit = async (values: RateLimitFields) => {
```

Comment:
Is this still needed as we have the updateAgentMutation now?

Reply:
this is still needed, as we dont want to override all the interfaceConfig, and the updateAgentMutation does not allow to update singular fields of a JSON

Reply:
I got rid of the RateLimitForm and its implementation, the AgentForm should be responsible for all the Agent fields, which now possible with the new update endpoint.